src/js/js-loader.ts

```diff
@@ -33,9 +33,9 @@ function jsLoader(source: string, sourceMaps: any): void {
 }
 
 function extractTranslations(loader: TranslateLoaderContext, source: string, sourceMaps: any): void {
-    const options = loaderUtils.parseQuery(loader.query);
+    const options = (loaderUtils as any).getOptions(loader) || {};
```

Comment:
We should update the typings of loader-utils to avoid a cast to any here. 

Therefore, add the following function in typings/loader-utils/loader-utils.d.ts

```js
    export function getOptions<T>(loaderContext: { query: string; }): T;
```

In addition, we have to change the `typings/webpack/webpack.plugin.d.ts` file and make the `query` property mandatory

```js
interface LoaderContext {
        version?: number;
        ....
        query: string;
        ....
}
```

In the `js-loader` a new interface needs to be created that defines the type for the options object (insert it at the top)

```js
/**
 * The optional options passed to the plugin
 */
interface LoaderOptions {
    /**
     * Optional acorn options that are passed to the parser
     */
    parserOptions?: acorn.Options;
}
```
Then we can change this --- and the following --- line to

```js
const options = loaderUtils.getOptions<LoaderOptions>(loader);
const parserOptions = options.parserOptions || {};
```